cfme/tests/cloud/test_quota.py

```diff
@@ -0,0 +1,326 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+from riggerlib import recursive_update
+
+from cfme import test_requirements
+from cfme.base.credential import Credential
+from cfme.cloud.provider.azure import AzureProvider
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.utils.generators import random_vm_name
+
+from widgetastic.utils import partial_match
+
+
+pytestmark = [
+    test_requirements.quota,
+    pytest.mark.long_running,
+    pytest.mark.usefixtures("setup_provider"),
+    pytest.mark.provider(
+        [AzureProvider, OpenStackProvider],
+        required_fields=[["provisioning", "image"]],
+        scope="function",
+    ),
+]
+
+
+def new_credential():
+    return Credential(principal='uid{}'.format(fauxfactory.gen_alphanumeric(4)),
+                      secret='redhat')
+
+
+@pytest.fixture
+def vm_name():
+    return random_vm_name(context="quota")
+
+
+@pytest.fixture
+def template_name(provisioning):
+    return provisioning["image"]["name"]
+
+
+@pytest.fixture
+def prov_data(appliance, provider, provisioning, vm_name):
+    if provider.one_of(OpenStackProvider):
+        return {
+            "catalog": {"vm_name": vm_name},
+            "environment": {"automatic_placement": True},
+            "properties": {"instance_type": partial_match("m1.large")},
+        }
+    if provider.one_of(AzureProvider):
+        instance_type = "d2s_v3" if appliance.version < "5.10" else "D2s_v3"
+        return {
+            "catalog": {"vm_name": vm_name},
+            "environment": {"automatic_placement": True},
+            "properties": {"instance_type": partial_match(instance_type)},
+            "customize": {
+                "admin_username": provisioning["customize_username"],
+                "root_password": provisioning["customize_password"],
+            },
+        }
+
+
+@pytest.fixture
+def set_child_tenant_quota(request, appliance, new_child):
+    """This fixture assigns quota to child tenant"""
+    field, value = request.param
+    new_child.set_quota(**{"{}_cb".format(field): True, field: value})
+    yield
+    # will refresh page as navigation to configuration is blocked if alerts are on the page
+    appliance.server.login_admin()
+    appliance.server.browser.refresh()
```

Comment:
Question:
why you need browser update after login?. Is there any error you found?

Reply:
- As this test case was giving ```TimedoutError``` if I did not use ```appliance.server.login_admin()```. 
- ```appliance.server.browser.refresh()``` used for just refreshing browser. I did not get error related to it.
- Now, I tried with removing login via ``` appliance.server.login_admin()``` and test case executed successfully.